additional_data_sources/siren-function-source/scripts/import_siren.py

```diff
@@ -0,0 +1,192 @@
+                    "siren": result["unitesLegales"][i]["siren"],
+                    "unite_legale": "Not found",
+                    "derniertraitement": "Not found",
+                    "dateCreationUniteLegale": "Not found",
+                    "identifiantAssociationUniteLegale": "Not found",
+                    "trancheEffectifsUniteLegale": "Not found",
+                    "anneeEffectifsUniteLegale": "Not found",
+                    "dateDernierTraitementUniteLegale": "Not found",
+                    "categorieEntreprise": "Not found",
+                    "etatAdministratifUniteLegale": "Not found",
+                    "nomUniteLegale": "Not found",
+                    "denominationUniteLegale": "Not found",
+                    "categorieJuridiqueUniteLegale": "Not found",
+                    "activitePrincipaleUniteLegale": "Not found",
+                    "changementCategorieJuridiqueUniteLegale": "Not found",
+                    "nomenclatureActivitePrincipaleUniteLegale": "Not found",
+                    "economieSocialeSolidaireUniteLegale": "Not found",
+                    "caractereEmployeurUniteLegale": "Not found",
+                }
+            )
+
+            pass
+
+    return siren_info_list
+
+
+# put token secrets
+def query_siren():
+    siren_info_list = []
+    headers = {
+        "Accept": "application/json",
+        "Authorization": f"""Bearer {TOKEN}""",
+    }
+    siren_list = get_offerer_siren_list()
+    if siren_list.shape[0] > MAX_SIREN_CALL:
+        nb_df_sub_divisions = siren_list.shape[0] // MAX_SIREN_CALL
+        for k in range(nb_df_sub_divisions):
+            temp_siren_list = siren_list[k * MAX_SIREN_CALL : (k + 1) * MAX_SIREN_CALL]
+            temp_siren_list.reset_index(drop=True, inplace=True)
+            query = get_siren_query(temp_siren_list["siren"])
+            response = requests.get(
+                query,
+                headers=headers,
+            )
+            if response.status_code != 200:
+                print("error")
+            result = response.json()
+            siren_info_list = append_info_siren_list(siren_info_list, result)
+            time.sleep(2)
+        temp_siren_list = siren_list[nb_df_sub_divisions * MAX_SIREN_CALL :]
+        temp_siren_list.reset_index(drop=True, inplace=True)
+        query = get_siren_query(temp_siren_list["siren"])
+        response = requests.get(
+            query,
+            headers=headers,
+        )
+        if response.status_code != 200:
+            print("error")
+        result = response.json()
+        siren_info_list = append_info_siren_list(siren_info_list, result)
+    else:
+        query = get_siren_query(siren_list["siren"])
+        response = requests.get(
+            query,
+            headers=headers,
+        )
+        if response.status_code != 200:
+            print("error")
+        result = response.json()
+        siren_info_list = append_info_siren_list(siren_info_list, result)
```

Comment:
Tu peux aussi le mettre dans la boucle: si c'est < MAX : t'auras nb_df_sub_divisions=0.
Sauf que en mettant for k in range(nb_df_sub_divisions + 1) ça prendra bien ce cas et celui d'avant